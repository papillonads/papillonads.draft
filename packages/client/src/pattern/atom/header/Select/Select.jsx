import React from 'react'
import cn from 'classnames'
import cs from './Select.scss'
import uuidv1 from 'uuid/v1'
import { selectDefaultProps, selectPropTypes } from './Select.props'

const Select = ({ className, placeholder, items }) => {
  const selectClassList = cn(className, cs.select)

  return (
    <React.Fragment>
      <select className={selectClassList}>
        <option className={cs.selectOption} key={uuidv1()} value={placeholder}>
          {placeholder}
        </option>
        {items.map(item => (
          <option key={uuidv1()} value={item.text}>
            {item.text}
          </option>
        ))}
      </select>
    </React.Fragment>
  )
}

Select.defaultProps = selectDefaultProps

Select.propTypes = selectPropTypes

export default Select
