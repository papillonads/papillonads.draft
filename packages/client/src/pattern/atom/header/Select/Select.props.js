import { string, arrayOf, shape } from 'prop-types'

export const selectDefaultProps = {
  className: '',
  placeholder: '',
}

export const selectPropTypes = {
  className: string,
  placeholder: string,
  items: arrayOf(
    shape({
      text: string.isRequired,
    }),
  ).isRequired,
}
