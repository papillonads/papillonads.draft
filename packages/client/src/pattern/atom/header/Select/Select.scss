.select {
  padding: 8px;
  // -moz-appearance: window;
  // -webkit-appearance: none;
}
