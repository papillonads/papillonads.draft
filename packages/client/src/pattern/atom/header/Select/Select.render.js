import React from 'react'
import Select from './Select'

export const items = [{ text: 'item 1' }, { text: 'item 2' }, { text: 'item 3' }]

const defaultRender = <Select items={items} />

export { defaultRender }
