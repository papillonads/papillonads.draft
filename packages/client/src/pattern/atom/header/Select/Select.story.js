import { storiesOf } from '@storybook/react'
import { defaultRender } from './Select.render'
import { getInfo } from '../../../../../../../.storybook/utility'

const defaultRenderInfo = getInfo('default')

storiesOf('Patttern/Atom/Header/Select', module).add(defaultRenderInfo.text, () => defaultRender, defaultRenderInfo.parameters)
