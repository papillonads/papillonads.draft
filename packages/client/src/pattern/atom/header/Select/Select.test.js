import { defaultRender } from './Select.render'

describe('<Select />', () => {
  describe('Snaphot', () => {
    test('must match defaultRender', () => {
      expect(global.renderToJSON(defaultRender)).toMatchSnapshot()
    })
  })
})
