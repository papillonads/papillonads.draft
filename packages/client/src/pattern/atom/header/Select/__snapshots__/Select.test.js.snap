// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> Snaphot must match defaultRender 1`] = `
<select
  className=""
>
  <option
    value=""
  >
    
  </option>
  <option
    value="item 1"
  >
    item 1
  </option>
  <option
    value="item 2"
  >
    item 2
  </option>
  <option
    value="item 3"
  >
    item 3
  </option>
</select>
`;
